esp/esp/utils/latex.py

```diff
@@ -170,8 +170,17 @@ def _gen_latex(texcode, stdout, stderr, type='pdf'):
     elif retcode:
         # Otherwise, if there was an error, we want to exit now since things
         # didn't work.
-        # TODO(benkraft): Return part of the log file here.
-        raise ESPError("LaTeX failed; try looking at the log file.")
+        # TODO(benkraft): Try to extract the actual error out.
+        raise ESPError('LaTeX failed; try looking at the log file.  Here are '
+                       'the last 1000 characters of the log: %s'
+                       % tex_log[-1000:])
```

Comment:
Does this display okay? I haven't looked at it myself so maybe it's fine, but I'd be concerned about admins missing the message at the top and getting confused by the error spew.